### ISSUE TITLE
Add accessible title to HelpText when label is ReactNode

### ISSUE THREAD

**Issue by mikaelrss** [digdir/designsystem#576]
### Description

We would like to have support for providing a `ReactNode` as `label` for `RadioButtons`. Currently, the `RadioButton` uses the passed `label` prop as a `title` for the `HelpText` component. If `label` is anything other than a `string`, `HelpText` will ignore the title completely. Therefore, we are limited to having to pass only string values as `label`. 

So we would like a way to provide a `label` that is `ReactNode`, and still have a `HelpText` with a correct `title`. This could for instance be solved with a `accessibleHelpTextTitle`-prop that can be used if `label` is not a `string`. Or possibly with a composable `RadioButton`: 

```jsx
<RadioButton>
  <RadioButton.Label>...</RadioButton.Label>
  <RadioButton.HelpText>...</RadioButton.HelpText>
</RadioButton>
```

**Comment by framitdavid** [comment]
@mikaelrss, thanks for your feature request. I need some clarifications to ensure I understand this issue correctly. 

1. Is this a description which should be referenced with aria-describedBy? 🤔
2. What's the use case to provide the ReactNode instead of a string? (Do you have an example?)

Thanks! 😎

**Comment by mikaelrss** [comment]
@framitdavid Sure!

The need comes from the `RadioButton` we have below. We would like to be able to hightlight the persons name in bold. The way to do that in apps today is to provide a text resource with markdown syntax specifying that the text should be bold. 
![image](https://github.com/digdir/designsystem/assets/26043795/48fc2785-0a3f-4b36-990a-4cd5cf4cb53e)


So we want to be able to do this: 

```jsx
<RadioButton label={<strong>I want my label to be bold text</strong>} />
```
And this is supported today. However, whenever the `label` prop is not a string but rather a `ReactNode`, todays implementation removes the `HelpText` title completely with this implementation:

```jsx
<HelpText
  size={size}
  title={typeof label === "string" ? `Help text for ${label}` : ""}
>
  {helpText}
</HelpText>
```
So we want to be able to pass a `React.ReactNode` as `label`, *and* still be able to provide a accessible text to use as title for the `HelpText` component. Could be solved by providing an additional prop for instance.

**Comment by framitdavid** [comment]
@mikaelrss thanks for more information about this. I suggest that we try to achieve this without doing changes within the FDS. The RadioButton supports the className prop, hence we can achieve bold text by providing className.

```
<RadioButton className="customLabel">....</RadioButton>

.customLabel label {
   font-weight: bold; <--- otherwise, use a token from FDS.
}

```

What do you think?

**Comment by mikaelrss** [comment]
For apps though, there is no possibility for us to provide styling to the component. A `RadioButton` in apps is only given a text resource, and as far as I know, no way of styling this text resource. The way this is done is by supporting markdown for the given text resource.

Styling with `classNames`  would solve the issue of a11y title missing whenever somebody passes a `ReactNode` label to `HelpText` either. 

@framitdavid

**Comment by framitdavid** [comment]
Just to clarify my understanding. What about avoiding markdown for the label within Apps for the RadioCardComponent? Should not be necessary to provide a markdown when the RadioCardComponent internally is handling the styling for bold labels? 🤔 😄 Then we also ensure that every RadioCardComponent within the app has the same visual expression.

**Comment by mikaelrss** [comment]
That could be a solution, but then we would have to add an option for bold text in apps. :sweat_smile: This gives very little flexibility, and would need a specific option for each and every styling change an app would like to implement. The advantage of using markdown, is that text can be formatted in a very flexible way for all apps. This is also how text is formatted in all other components, including the description field of the `RadioButtons`. It is an API the owners of apps are familiar with. 

The `RadioButton` in the design system already supports the behavior, the problem is only that it ignores the a11y of the `HelpText` when using this option. Do you want to remove the possibility of passing `ReactNode` as a label completely and force users to only pass plain strings?

**Comment by framitdavid** [comment]
Thanks for the feedback, @mikaelrss! 😄 

I was genuinely seeking a deeper understanding and aiming to critically examine the proposed solution. Issue #524 is all about creating a brand new radio button component for FDS. I assigned this issue to myself today to get started on the implementation. Hence I needed more details about your challenges. 

I will follow up on this feature request and keep you posted. Thanks.

**Comment by framitdavid** [comment]
@mikaelrss I move this issue to the backlog - but it will be fixed within #524 :)

**Comment by mimarz** [comment]
Could the new approach to `Radio` component be solve this for you @mikaelrss ? Have a look at the new component api in #666 :)

**Comment by mikaelrss** [comment]
@mimarz Yes. I think this would solve our issue. The new `Radio`-component does not render a `HelpText`, which would give us the control we need to set our own accessible title to a `HelpText`. 

We did however agree that for now, we should not support `ReactNode` as `title` for `Radio`-components, so the problem does not really apply anymore (for now).
